syntax = "proto3";

package calc;

service Calculator {

	rpc Add (AddRequest) returns (AddReply) {}
	rpc Substract (SubstractRequest) returns (SubstractReply) {}
	rpc Multiply (MultiplyRequest) returns (MultiplyReply) {}
	rpc Divide (DivideRequest) returns (DivideReply) {}
}

message AddRequest{
	
	int32 n1=1;
	int32 n2=2;
}

message AddReply{
	
	int32 n1=1;
}

message SubstractRequest{
	
	int32 n1=1;
	int32 n2=2;
}

message SubstractReply{
	
	int32 n1=1;
}

message MultiplyRequest{
	
	int32 n1=1;
	int32 n2=2;
}

message MultiplyReply{
	
	int32 n1=1;
}

message DivideRequest{
	
	int32 n1=1;
	int32 n2=2;
}

message DivideReply{
	
	float f1=1;
}
